src/main/java/org/snomed/snowstorm/rest/MultiSearchController.java

```diff
@@ -75,7 +78,71 @@ public ItemsPage<BrowserDescriptionSearchResult> findDescriptions(
+		timer.checkpoint("description search");
+
+		Map<String, List<Description>> branchDescriptions = new HashMap<>();
+		Map<String, List<String>> branchConceptIds = new HashMap<>();
+		for (Description description : descriptions) {
+			branchDescriptions.computeIfAbsent(description.getPath(), s -> new ArrayList<>()).add(description);
+			branchConceptIds.computeIfAbsent(description.getPath(), s -> new ArrayList<>()).add(description.getConceptId());
+		}
+
+		List<LanguageDialect> languageDialects = ControllerHelper.parseAcceptLanguageHeaderWithDefaultFallback(acceptLanguageHeader);
+		Map<String, ConceptMini> conceptMiniMap = new HashMap<>();
+		for (String branchPath : branchConceptIds.keySet()) {
+			conceptMiniMap.putAll(conceptService.findConceptMinis(branchPath, branchConceptIds.get(branchPath), languageDialects).getResultsMap());
+			timer.checkpoint("Join concepts from " + branchPath);
+		}
+
+		List<BrowserDescriptionSearchResult> results = new ArrayList<>();
+		for (Description description : descriptions) {
+			BrowserDescriptionSearchResult result = new BrowserDescriptionSearchResult(description.getTerm(), description.isActive(), description.getLanguageCode(),
+					description.getModuleId(), conceptMiniMap.get(description.getConceptId()));
+			result.addExtraField("branchPath", description.getPath());
+			results.add(result);
+		}
+		timer.finish();
+
+		return new ItemsPage<>(new PageImpl<>(results, pageRequest, descriptions.getTotalElements()));
+	}
+	
+	@Operation(description = "Search descriptions across multiple Code Systems. Can search specified branches in addition to all published branches.")
+	@PostMapping(value = "multisearch/descriptions", produces = {"application/json", "text/csv"})
+	@JsonView(value = View.Component.class)
+	public ItemsPage<BrowserDescriptionSearchResult> findDescriptionsIncludingRequestedBranches(
+			@RequestParam String term,// Required!
+			@RequestParam(required = false) Boolean active,
+			@RequestParam(required = false) Collection<String> module,
+			@RequestParam(required = false) String ecl,
+
+			@Parameter(description = "Set of two character language codes to match. " +
+					"The English language code 'en' will not be added automatically, in contrast to the Accept-Language header which always includes it. " +
+					"Accept-Language header still controls result FSN and PT language selection.")
+			@RequestParam(required = false) Set<String> language,
+
+			@Parameter(name = "Set of description types to include. Pick descendants of '900000000000446008 | Description type (core metadata concept) |'.")
+			@RequestParam(required = false) Set<Long> type,
+
+			@RequestParam(required = false) Boolean conceptActive,
+			@RequestParam(defaultValue = "ALL_PUBLISHED_CONTENT") ContentScope contentScope,
+			@RequestParam(defaultValue = "0") int offset,
+			@RequestParam(defaultValue = "50") int limit,
+			@RequestBody MultibranchDescriptionSearchRequest includeBranches,
```

Comment:
@RichardAWood What is the main reason for creating this POST api? Is it because you need post  a large set of branches to include? Also when includeBranches are set do you still need published branches included as well?

Reply:
@CoderMChu - this was created in collaboration with @pgwilliams because the existing api's didn't return non-published branches, which is required by RT2.  You can look at more history in JIRA ticket RT2-1567, or in the original PR for this change: PR #489.

Reply:
@RichardAWood You could use existing GET api by adding @RequestParam(required = false) Set<String> branches. @kaicode What do you think?

Reply:
Yeah, @pgwilliams and I discussed that option when we were developing this, but unfortunately the number of branches we need to be able to pass into this call exceeds the character limit for it to be a RequestParam.  That's the main reason we went with setting up the new POST api.